smartsim/_core/_cli/utils.py

```diff
@@ -56,28 +57,50 @@ def color_bool(trigger: bool = True) -> str:
     return colorize(str(trigger), color=_color)
 
 
-def pip_install(packages: t.List[str], end_point: t.Optional[str] = None, verbose: bool = False) -> None:
+def pip_install(
+    packages: t.List[str], end_point: t.Optional[str] = None, verbose: bool = False
+) -> None:
     """Install a pip package to be used in the SmartSim build
     Currently only Torch shared libraries are re-used for the build
     """
     # form pip install command
-    cmd = ["python", "-m", "pip", "install"]
-    cmd.extend(packages)
+    cmd = [sys.executable, "-m", "pip", "install"]
+
     if end_point:
-        cmd.extend(["-f", end_point])
+        cmd.extend(("-f", str(end_point)))
 
-    cmd_arg = " ".join(cmd)
+    cmd.extend(packages)
 
     if verbose:
-        logger.info(f"Installing packages {packages}...")
+        logger.info(f"Installing packages: {', '.join(packages)}")
     proc = subprocess.Popen(
-        cmd_arg, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE
     )
     _, err = proc.communicate()
     returncode = int(proc.returncode)
     if returncode != 0:
-        error = f"{packages} installation failed with exitcode {returncode}\n"
-        error += err.decode("utf-8")
+        error = (
+            f"'{' '.join(cmd)}' installation failed with exitcode {returncode}\n"
```

Comment:
It's cool to be able to kick off all the installations with a single command, but it would be more useful for users to be able to tell them which package failed to install

Reply:
Agreed!! Changed!